samples/Basecamp.gs

```diff
@@ -0,0 +1,65 @@
+var CLIENT_ID = '...';
+var CLIENT_SECRET = '...';
+
+/**
+ * Authorizes and makes a request to the Basecamp 3 API.
+ */
+function run() {
+  var service = getService();
+  if (service.hasAccess()) {
+    var url = 'https://launchpad.37signals.com/authorization.json';
+    var response = UrlFetchApp.fetch(url, {
+      headers: {
+        Authorization: 'Bearer ' + service.getAccessToken()
+      }
+    });
+    var result = JSON.parse(response.getContentText());
+    Logger.log(JSON.stringify(result, null, 2));
+  } else {
+    var authorizationUrl = service.getAuthorizationUrl();
+    Logger.log('Open the following URL and re-run the script: %s',
+        authorizationUrl);
+  }
+}
+
+/**
+ * Reset the authorization state, so that it can be re-tested.
+ */
+function reset() {
+  var service = getService();
+  service.reset();
+}
+
+/**
+ * Configures the service.
+ */
+function getService() {
+  return OAuth2.createService('Basecamp')
+      // Set the endpoint URLs.
+      .setAuthorizationBaseUrl('https://launchpad.37signals.com/authorization/new?type=web_server')
```

Comment:
I would avoid adding this in. These samples are meant to show how to configure the library with the given OAuth provider, but they shouldn't get into the details of how to register an application, get keys, etc. Those details are best left to the provider's own documentation, where it is more likely to stay up-to-date. Perhaps we just link off to their documentation?